packages/frontend/src/main/components/common/SpeckleViewer.vue

```diff
@@ -93,10 +99,23 @@ export default {
       })
 
       this.viewer.on(
-        'load-progress',
-        throttle((args) => this.$emit('load-progress', args), 250)
+        ViewerEvent.LoadProgress,
+        throttle((args) => this.$emit(ViewerEvent.LoadProgress, args), 250)
       )
-      this.viewer.on('select', (objects) => this.$emit('selection', objects))
+
+      this.viewer.on(ViewerEvent.ObjectClicked, (selectionInfo) => {
+        handleViewerSelection(selectionInfo)
+      })
+
+      this.viewer.on(ViewerEvent.ObjectDoubleClicked, (selectionInfo) => {
+        handleViewerDoubleClick(selectionInfo)
+      })
+
+      window.addEventListener('keydown', (e) => {
```

Comment:
event listeners should be cleaned up https://github.com/specklesystems/speckle-server/pull/977/files#r961520356